src/view/components/customerIdWrapper.jsx

```diff
@@ -0,0 +1,186 @@
+import "regenerator-runtime"; // needed for some of react-spectrum
+import React, { useState, useEffect } from "react";
+import { FieldArray } from "formik";
+import Textfield from "@react/react-spectrum/Textfield";
+import Checkbox from "@react/react-spectrum/Checkbox";
+import Select from "@react/react-spectrum/Select";
+import FieldLabel from "@react/react-spectrum/FieldLabel";
+import Button from "@react/react-spectrum/Button";
+import Well from "@react/react-spectrum/Well";
+import Heading from "@react/react-spectrum/Heading";
+import Delete from "@react/react-spectrum/Icon/Delete";
+import ModalTrigger from "@react/react-spectrum/ModalTrigger";
+import Dialog from "@react/react-spectrum/Dialog";
+import "@react/react-spectrum/Form"; // needed for spectrum form styles
+import PropTypes from "prop-types";
+import WrappedField from "./wrappedField";
+import authenticatedStateOptions from "../constants/authenticatedStateOptions";
+import "../actions/setCustomerIds.styl";
+import getDefaultCustomerId from "../utils/getDefaultCustomerId";
+import ErrorMessage from "./errorMessage";
+
+function cidWrapper({ values }) {
+  // const [namespaceOptions, setNamespaceOptions] = useState([]);
+  const [showStuff, setShowStuff] = useState({});
```

Comment:
I recommend breaking this out into two pieces: 

```
const [identityNamespaces, setIdentityNamespaces] = useState();
const [error, setError] = useState();
```

Then in your `useEffect`, when you receive identity namespaces from the server, you can call `setIdentityNamespaces`. If you get an error, you can call `setError`. Then when rendering you can say if there's an error, show the error. Otherwise, if we can identity namespaces, show the content (which will display identity namespaces). Otherwise, show a loading indicator. Anyway, this can happen later when you actually implement namespace loading.